### PR TITLE
DT-1573 Report to show state of play of matching

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/reports/MatchSummaryReport.kt [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  return this.copy(
+    total = this.total + 1,
+    completed = this.completed + message,
+    waiting = this.waiting + message,
+    `exceeded-sla` = this.`exceeded-sla` + message,
+  )
+}
+
+operator fun Completed.plus(message: Message): Completed {
+  return message.takeIf { it.isProcessed() }?.let {
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/reports/MatchSummaryReport.kt:90]: Plus implies it is added - so maybe the takeIf should go up a level?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/reports/ReportsConfiguration.kt [hunk 2]
```diff
@@ -39,12 +40,19 @@ class ReportsConfiguration(
       beanClass = ProcessedReport::class,
       beanMethod = "generate"
     ),
+    RouterOperation(
+      path = "/report//match-summary",
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/reports/ReportsConfiguration.kt:44]: typo?

**Review by mikehalmamoj — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/reports/MatchSummaryReport.kt (hunk 1 above)
Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/reports/MatchSummaryReport.kt:90]: agreed, I'll do a new PR

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/reports/ReportsConfiguration.kt (hunk 2 above)
Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/reports/ReportsConfiguration.kt:44]: yep

**Pull request merged**